### PR TITLE
Add Heatmap plotting function and notebook for demonstration (#31)

### PR DESCRIPTION
**Pull request by jakobchwastek**
## Description

I implemented a heatmap function using `histogram2d` and `imshow` and a very very basic notebook. I used imshow cause i thought  it has better interpolation capabilities than `pcolormesh`. Maybe you have to correct me. 
Please have a look and tell me what needs to be adjusted.

Fixes issue #31

## Implemented changes

Insert a description of the changes implemented in the pull request.

- [x] Implement heatmap function
- [x] tutorial Notebook

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

This has just been tested by using the function in the tutorial notebook

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by dkrako — CHANGES_REQUESTED** [review]
This is very clean and has a clever way of using the experiment definition from the gaze dataframe.

I would just like to remove the `origin` argument from the plotting function.
this should be inferable from the `experiment.screen.origin` shouldn't it?

Is this related to the bug which you mentioned, where the origin should actually be `upper left` instead `lower left`?

File: src/pymovements/plotting/heatmap.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Heatmap module.
+"""
+import matplotlib.pyplot as plt
```
Comment (dkrako) [inline src/pymovements/plotting/heatmap.py:23]: you need an additional import to get rid of all the test errors:

```
from __future__ import annotations
```

**Review by dkrako — CHANGES_REQUESTED** [review]

**Comment by jakobchwastek** [comment]
> I would just like to remove the `origin` argument from the plotting function.
> this should be inferable from the `experiment.screen.origin` shouldn't it?
> 
> Is this related to the bug which you mentioned, where the origin should actually be `upper left` instead `lower left`?

Yes, but i think when data is loaded with the correct origin in the first place it is not necessary to use origin here at all. How I understand all data should have the same coordinates system when choosing the correct origin in the Experiment/Screen class.

So should I remove argument or leave it for now?

**Comment by dkrako** [comment]
> Looks like there is a problem because GazeDataFrame.experiment is optional..

Hmm.. okay then for this PR let's just raise an exception if experiment is None

**Comment by dkrako** [comment]
If mypy still doesn't get it, just add a line with `assert gaze.experiment is not None`

we will fix this in a future PR

**Review by dkrako — CHANGES_REQUESTED** [review]

File: src/pymovements/plotting/heatmap.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        ymin, ymax = screen.y_min_dva, screen.y_max_dva
+    else:
+        raise ValueError('Position columns must be in pixels or degrees')
+
+    # Define the grid and bin the gaze data
+    x_bins = np.linspace(xmin, xmax, num=gridsize[0])
+    y_bins = np.linspace(ymin, ymax, num=gridsize[1])
+
+    # Bin the gaze data
+    heatmap, x_edges, y_edges = np.histogram2d(x, y, bins=[x_bins, y_bins])
```
Comment (dkrako) [inline src/pymovements/plotting/heatmap.py:128]: naming the variable `heatmap` is problematic as it is also the function name.
our linters don't accept that.

name the variable `heatmap_values` or similar

**Review by dkrako — CHANGES_REQUESTED** [review]

**Review by dkrako — CHANGES_REQUESTED** [review]
for testing you can just copy the testing functions from here:
https://github.com/aeye-lab/pymovements/blob/main/tests/plotting/traceplot_test.py

then just adjust the function and parameter names accordingly to get 100% coverage.

**Review by dkrako — APPROVED** [review]
Thanks a lot! LGTM!

**Pull request merged**